controllers/common/helper.go

```diff
@@ -306,7 +306,21 @@ func ReconcileDBRestoreFromDataSource(
 	ctx context.Context,
 	c client.Client,
 	database *everestv1alpha1.DatabaseCluster,
+	clientGet func(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error,
 ) error {
+	dbr := &everestv1alpha1.DatabaseClusterRestore{}
+	err := clientGet(ctx, types.NamespacedName{
+		Namespace: database.Namespace,
+		Name:      database.Name,
+	}, dbr)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if dbr.IsComplete(database.Spec.Engine.Type) {
+		database.Spec.DataSource = nil
+		return nil
+	}
+
```

Comment:
I may be missing something here, but we can probably also skip the `IsNotFound` check, because on the caller side there's an early return if the DataSource is `nil`

Reply:
The first part regarding the client - right, thanks, updated.

The second part - I don't think it will work. 
We should first check if there is an existing restore completed and if so - cleanup the datasource and don't create another restore. Otherwise if we delete the restore object it will be recreated again which we don't want.

Reply:
regarding the `IsNotFound` - I think we should keep it. We check the restore object and if it's not there we should proceed with the next actions but not return an error  🤔